src/main/java/uk/ac/ebi/eva/pipeline/listeners/StepProgressListener.java

```diff
@@ -36,22 +36,24 @@ public void beforeChunk(ChunkContext context) {
     @Override
     public void afterChunk(ChunkContext context) {
 
-        long estimatedTotalNumberOfLines = (long)context.getStepContext().getStepExecutionContext()
+        Object lines = context.getStepContext().getStepExecutionContext()
```

Comment:
nope, execution context is a map<String, Object>, the casting is still required (even for Object to Long).

I could do the next to join the retrieval, default value and casting: 
```
long estimatedTotalNumberOfLines = (long) context.getStepContext().getStepExecutionContext()
    .getOrDefault(ExecutionContextParametersNames.NUMBER_OF_LINES, 0L);
```